src/creature.h

```diff
@@ -198,51 +206,51 @@ class Creature : virtual public Thing
 	virtual void onWalkComplete() {}
 
 	// follow functions
-	Creature* getFollowCreature() const { return followCreature; }
-	virtual void setFollowCreature(Creature* creature);
+	std::shared_ptr<Creature> getFollowCreature() const { return followCreature; }
+	virtual void setFollowCreature(std::shared_ptr<Creature> creature);
 	virtual void removeFollowCreature();
-	virtual bool canFollowCreature(Creature* creature);
-	virtual bool isFollowingCreature(Creature* creature) { return followCreature == creature; }
+	virtual bool canFollowCreature(std::shared_ptr<Creature> creature);
+	virtual bool isFollowingCreature(std::shared_ptr<Creature> creature) { return followCreature == creature; }
 
 	// follow events
-	virtual void onFollowCreature(const Creature*);
+	virtual void onFollowCreature(std::shared_ptr<const Creature>);
 	virtual void onUnfollowCreature();
 
 	// Pathfinding functions
-	bool isFollower(const Creature* creature);
-	void addFollower(Creature* creature);
-	void removeFollower(Creature* creature);
+	bool isFollower(std::shared_ptr<const Creature> creature);
+	void addFollower(std::shared_ptr<Creature> creature);
+	void removeFollower(std::shared_ptr<Creature> creature);
 	void removeFollowers();
 	void releaseFollowers();
 
 	// Pathfinding events
 	void updateFollowersPaths();
 
 	// combat functions
-	Creature* getAttackedCreature() { return attackedCreature; }
-	virtual void setAttackedCreature(Creature* creature);
+	std::shared_ptr<Creature> getAttackedCreature() { return attackedCreature; }
+	virtual void setAttackedCreature(std::shared_ptr<Creature> creature);
 	virtual void removeAttackedCreature();
-	virtual bool canAttackCreature(Creature* creature);
-	virtual bool isAttackingCreature(Creature* creature) { return attackedCreature == creature; }
+	virtual bool canAttackCreature(std::shared_ptr<Creature> creature);
+	virtual bool isAttackingCreature(std::shared_ptr<Creature> creature) { return attackedCreature == creature; }
 
-	virtual BlockType_t blockHit(Creature* attacker, CombatType_t combatType, int32_t& damage,
+	virtual BlockType_t blockHit(std::shared_ptr<Creature> attacker, CombatType_t combatType, int32_t& damage,
 	                             bool checkDefense = false, bool checkArmor = false, bool field = false,
 	                             bool ignoreResistances = false);
 
-	bool setMaster(Creature* newMaster);
+	bool setMaster(std::shared_ptr<Creature> newMaster);
 
 	void removeMaster()
 	{
 		if (master) {
 			master = nullptr;
-			decrementReferenceCounter();
 		}
 	}
```

Comment:
We don’t need to check the `master` here
```suggestion
	void removeMaster()
	{
			master = nullptr;
	}
```